README.rst

```diff
@@ -217,20 +217,29 @@ As of 3.0.0 the ``$2y$`` prefix is still supported in ``hashpw`` but deprecated.
 Maximum Password Length
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-The bcrypt algorithm only handles passwords up to 72 characters, any characters
+The bcrypt algorithm only handles passwords up to 72 characters; any characters
 beyond that are ignored. To work around this, a common approach is to hash a
-password with a cryptographic hash (such as ``sha256``) and then base64
-encode it to prevent NULL byte problems before hashing the result with
+password with a keyed cryptographic hash (such as ``bcrypt_pbkdf``) and then
+base64 encode it to prevent NULL byte problems before hashing the result with
 ``bcrypt``:
 
 .. code:: pycon
 
+    >>> import base64
+    >>> import bcrypt
     >>> password = b"an incredibly long password" * 10
     >>> hashed = bcrypt.hashpw(
-    ...     base64.b64encode(hashlib.sha256(password).digest()),
+    ...     base64.b64encode(bcrypt.kdf(password=password,
+    ...                                 salt=pepper,
+    ...                                 desired_key_bytes=32,
+    ...                                 rounds=100)),
     ...     bcrypt.gensalt()
     ... )
 
+Using a hash function without a hash is `recommended against`_ as it may expose
```

Comment:
(without a salt)

Reply:
Feedback implemented in ee8a991fbb837f76c7711a00dc63fab293d00bfc. Let me know what you think :)